api/pkg/service/resource/resource.go

```diff
@@ -471,3 +466,10 @@ func invalidKindError(kind string) error {
 	return resource.MakeInvalidKind(fmt.Errorf("resource kind '%s' not supported. Supported kinds are %v",
 		kind, parser.SupportedKinds()))
 }
+
+func isNotFoundError(err error) error {
```

Comment:
imho, this doesn't improve the readabilty. 
One would `isNotFoundError(err)` to return true if the `err` is `NotFound` and not a `notFoundError` or `fetchError`. 
Moreover, it fails to log.Error for all errors other than `NotFound`